python/orca/example/learn/pytorch/brainMRI/README.md

```diff
@@ -1,59 +1,78 @@
 
 
-## Trouble Shooting
-- Difficult to identify a customized loss function; for example, dice coefficient
-    - Require a loss function creator to pass a customized loss function into orca
-- Unable to see the plot of matplotlib
-- Need to implement function to download dataset if it's not included in pytorch datasets
-
-## TODO
-- [ ] bigdl backend development
-- [ ] yarn development
-
 ## Prepare the environment
 
 We recommend you to use Anaconda to prepare the environment, especially if you want to run on a yarn cluster:
 
 ```
-conda create -n zoo python=3.7  # "zoo" is conda environment name, you can use any name you like.
-conda activate zoo
+conda create -n bigdl python=3.7  # "bigdl" is conda environment name, you can use any name you like.
+conda activate bigdl
 pip install torch
 pip install torchvision
+pip install albumentations
+pip install scikit-learn
+pip install opencv-python
 pip install matplotlib
 
-# For bigdl backend:
-pip install analytics-zoo  # 0.10.0.dev3 or above
-pip install jep==3.9.0
-pip install six cloudpickle
-
 # For torch_distributed backend:
-pip install analytics-zoo[ray]  # 0.10.0.dev3 or above
+pip install bigdl-orca[ray]
 
-# For spark backend
+# For spark backend:
 pip install bigdl-orca
 ```
 
-## Run on local after pip install
-
-`bigdl` backend is still developing.
+## Prepare the dataset
+You can download the dataset [here](https://www.kaggle.com/datasets/mateuszbuda/lgg-mri-segmentation/download).
 
-You can run with `torch_distributed` backend via:
 
 ```
-python brainMRI.py --backend torch_distributed
+# You can download the dataset to your project path.
+unzip archive.zip
+
+# if you use yarn-client, the dataset should be sent to the excutors. 
+# So we need zip the dataset as `.zip` file, then the program will send it to all excutors automatically. 
+cd kaggle_3m
+zip ../kaggle_3m.zip *
 ```
 
-You can run with `spark` backend via:
 
+You can run the example on local and yarn client mode.
+### Run on local after pip install
+
+The default backend is `torch_distributed`
+```commandline
+python brainMRI.py
 ```
-python brainMRI.py --backend spark
+You can run with `spark` backend via:
+```
+python brainMRI.py --backend spark 
 ```
 
-## Run on yarn cluster for yarn-client mode after pip install
-
+### Run on yarn cluster for yarn-client mode after pip install
 ```
 export HADOOP_CONF_DIR=the directory of the hadoop and yarn configurations
 python brainMRI.py --cluster_mode yarn-client
-```
\ No newline at end of file
+```
+
+If you use the spark as backend, you should give the `model_dir` parameter with the hdfs path. 
+
+Note: if you meet `OSError: Unable to load libhdfs: ./libhdfs.so: cannot open shared object file: No such file or directory` error, read the [document](https://bigdl.readthedocs.io/en/latest/doc/Orca/Overview/known_issues.html#oserror-unable-to-load-libhdfs-libhdfs-so-cannot-open-shared-object-file-no-such-file-or-directory).
```

Comment:
if -> If
remove error
read the document -> please refer to here for the solution